From 4a2114009ffe7fc54d3ae1556b2e84096635ea98 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 17 May 2021 10:04:07 +0200
Subject: [PATCH 2/4] Call the C++ DLL libLIEF.dll so it does not clash with
 lief.pyd

---
 CMakeLists.txt | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d00f7fa..a91b5f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -489,8 +489,18 @@ if(MSVC)
   else()
     set(CRT_OUTPUT_SUFFIX ${LIEF_USE_CRT_RELEASE})
   endif()
-  set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME LIEF${CRT_OUTPUT_SUFFIX}
-                                            CLEAN_DIRECT_OUTPUT 1)
+  # Since the Python extension module will be called lief.pyd, we cannot have the C++ Library
+  # called LIEF.dll since Windows will not load both.
+  if(WIN32 AND BUILD_SHARED_LIBS)
+    set_target_properties(
+      LIB_LIEF
+      PROPERTIES OUTPUT_NAME libLIEF${CRT_OUTPUT_SUFFIX})
+  else()
+    set_target_properties(
+      LIB_LIEF
+     PROPERTIES OUTPUT_NAME LIEF${CRT_OUTPUT_SUFFIX})
+  endif()
+  set_target_properties(LIB_LIEF PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 else()
   set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME LIEF CLEAN_DIRECT_OUTPUT
                                                              1)
-- 
2.30.1 (Apple Git-130)

