From 7ca3eb0b81b1f3a69dd713939ba3c93f64a2b880 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 9 Dec 2019 15:52:14 +0100
Subject: [PATCH 2/3] Add creation of BuildToolVersion

---
 api/python/MachO/pyMachO.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/api/python/MachO/pyMachO.cpp b/api/python/MachO/pyMachO.cpp
index 2a48172..ef9b422 100644
--- a/api/python/MachO/pyMachO.cpp
+++ b/api/python/MachO/pyMachO.cpp
@@ -66,6 +66,7 @@ void init_objects(py::module& m) {
   CREATE(DyldEnvironment, m);
   CREATE(EncryptionInfo, m);
   CREATE(BuildVersion, m);
+  CREATE(BuildToolVersion, m);
 }
 
 }
-- 
2.23.0

