From 40050c6259c73937ab6e7cb27dc8f90e2f79297a Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 17 May 2021 16:26:53 +0200
Subject: [PATCH 5/5] Add boost::throw_exception implementation

---
 src/errors.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/errors.cpp b/src/errors.cpp
index 93d067f..2657838 100644
--- a/src/errors.cpp
+++ b/src/errors.cpp
@@ -45,3 +45,15 @@ const std::error_category& lief_error_category() {
 LIEF::error_t LIEF::return_error(lief_errors e) {
   return boost::leaf::new_error(e);
 }
+
+
+// We need to explicitly define this here as Boost infers from the defines/environment
+// that boost::throw_exception will be user-defined.
+namespace boost
+{
+#ifdef BOOST_NO_EXCEPTIONS
+void throw_exception( std::exception const & e ){
+    throw 11;
+};
+#endif
+}// namespace boost
-- 
2.30.1 (Apple Git-130)

