From a4580d44441bede6ff8ea9958093c1cf322bc08c Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 17 May 2021 10:06:42 +0200
Subject: [PATCH 3/4] Add LIEFOptions for BUILD_STATIC_LIBS and
 BUILD_SHARED_LIBS

---
 cmake/LIEFOptions.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/LIEFOptions.cmake b/cmake/LIEFOptions.cmake
index 14b2b2dc..bbb26d24 100644
--- a/cmake/LIEFOptions.cmake
+++ b/cmake/LIEFOptions.cmake
@@ -28,6 +28,9 @@ option(LIEF_MACHO          "Build LIEF with MachO module"               ON)
 option(LIEF_DEX            "Build LIEF with DEX module"                 ON)
 option(LIEF_ART            "Build LIEF with ART module"                 ON)
 
+option(BUILD_STATIC_LIBS "Build LIEF (and pyLIEF mostly) statically"    OFF)
+option(BUILD_SHARED_LIBS "Build LIEF (and pyLIEF) shared"               ON)
+
 # OAT support relies on the ELF and DEX format.
 # Therefore, these options must be enabled to support use this format
 cmake_dependent_option(LIEF_OAT "Build LIEF with OAT module" ON
-- 
2.37.1

