From 8351f9a18795db8047a77d496fe4a5a5d5bf43c9 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 17 May 2021 16:26:53 +0200
Subject: [PATCH 4/4] Add boost::throw_exception implementation

---
 src/errors.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/errors.cpp b/src/errors.cpp
index bd2fbaf3..dfb262eb 100644
--- a/src/errors.cpp
+++ b/src/errors.cpp
@@ -49,3 +49,15 @@ const std::error_category& lief_error_category() {
 LIEF::error_t LIEF::return_error(lief_errors e) {
   return boost::leaf::new_error(e);
 }
+
+
+// We need to explicitly define this here as Boost infers from the defines/environment
+// that boost::throw_exception will be user-defined.
+namespace boost
+{
+#ifdef BOOST_NO_EXCEPTIONS
+void throw_exception( std::exception const & e ){
+    throw 11;
+};
+#endif
+}// namespace boost
-- 
2.37.1

